shared/lib_irradproc.cpp

```diff
@@ -1134,11 +1162,9 @@ int irrad::calc()
 	else
 	{	
 		// sun is down, assign sundown values
-		sunAnglesRadians[0] = -999*DTOR; //avoid returning a junk azimuth angle (return in radians)
-		sunAnglesRadians[1] = -999*DTOR; //avoid returning a junk zenith angle (return in radians)
-		sunAnglesRadians[2] = -999*DTOR; //avoid returning a junk elevation angle (return in radians)
-		timeStepSunPosition[0] = 0;
-		timeStepSunPosition[1] = 0;
+		solarpos(year, month, day, hour, minute, latitudeDegrees, longitudeDegrees, timezone, sunAnglesRadians);
+		timeStepSunPosition[0] = hour;
+		timeStepSunPosition[1] = (int)minute;
```

Comment:
If we are in continuous days, then starting at 1109, t_sunrise = -1 and t_sunset = 25, so then we will get t_sunset = 25-24 = 1, and in line 1120, t_sunset = 25. Then for the entire day, we are in line 1157 sunup case. So this looks right. 

In continuous nights, we have t_sunrise = 25 and t_sunset = -1, then we will go to line 1165 so that makes sense too.

But when we are in the last night of continous days, say sunset is hr 20 but will sunrise then be -1 from solarpos?

If so then, when we get from line 1120, t_sunset = 20+24, effectively pushing it back day after day until sunrise is no longer calculated as -1.

Reply:
Continuous nights won't trigger lines 1109 and 1116 because t_sunrise = 25 !< 0 (line 1109) and t_sunset = -1 !> 24 (line 1116). This will correctly trigger nighttime per line 1155 because t_cur will never be < -1 or > 25. 

Just to be safe, I'm also updating 225-232 to set sunrise/sunset to +- 100 on continuous nights/continuous days, so that even if they get rolled by 24 hrs, they'll still be out of the range of 0-24. 

In adding the check you suggested to check that sunrise/sunset from the next/previous day are also actually <0 or >24, this will fix the first/last day of continuous days problem. Good catch!